### PR TITLE
Support release image generation on Linux

### PR DESCRIPTION
**Pull request by jrtc27**
Figured it's best to get these into CheriBSD and upstream later/in parallel rather than being blocked on re@ reviews.

### CODE REVIEW

File: release/amd64/mkisoimages.sh [hunk 1]
```diff
@@ -112,7 +112,11 @@ if [ "$bootable" != "" ]; then
 
 	# Create a GPT image containing the partitions we need for hybrid boot.
 	hybridfilename=$(mktemp /tmp/hybrid.img.XXXXXX)
-	imgsize=`stat -f %z "$NAME"`
+	if [ "$(uname -s)" = "Linux" ]; then
```
Comment (brooksdavis) [inline release/amd64/mkisoimages.sh:115]: I'd be slightly tempted to use a case statement, but this is certainly fine.

**Review by brooksdavis — APPROVED** [review]

File: Makefile.inc1 [hunk 2]
```diff
@@ -1036,12 +1036,16 @@ WORLDTMP_MTREEFLAGS=	-deUW
...
 .endif
-DISTR_MTREE=	${MTREE_CMD}
+.if defined(DB_FROM_SRC)
+DISTR_MTREEFLAGS=	-N ${.CURDIR}/etc
+.endif
+DISTR_MTREECMD=	${MTREE_CMD}
 .if ${BUILD_WITH_STRICT_TMPPATH} != 0
-DISTR_MTREE=	${WORLDTMP}/legacy/usr/sbin/mtree
+DISTR_MTREECMD=	${WORLDTMP}/legacy/usr/sbin/mtree
 .endif
```
Comment (bsdjhb) [inline Makefile.inc1:1045]: I might have just used a += on DISTR_MTREE moving it after the BUILD_WITH_STRICT_TMPPATH as the diff would be slightly smaller, but this is fine.

**Review by bsdjhb — APPROVED** [review]

File: Makefile.inc1 (hunk 2 above)
Comment (jrtc27) [reply Makefile.inc1:1045]: That breaks the other _MTREE variables defined below, no? They’re not :=

Comment (jrtc27) [reply Makefile.inc1:1045]: Or at least it changes them, whether it matters, and whether they already have -N in that case, I don’t know

**Pull request merged**